src/Containers/Microsoft.NET.Build.Containers/ImageIndexGenerator.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.NET.Build.Containers.Resources;
+using Microsoft.NET.Build.Containers.Tasks;
+
+namespace Microsoft.NET.Build.Containers;
+
+internal struct ImageInfo
+{
+    internal string Config { get; init; }
+    internal string ManifestDigest { get; init; }
+    internal string Manifest { get; init; }
+    internal string ManifestMediaType { get; init; }
+}
+
+internal static class ImageIndexGenerator
+{
+    /// <summary>
+    /// Generates an image index from the given images.
+    /// </summary>
+    /// <param name="imageInfos"></param>
+    /// <returns>Returns json string of image index and image index mediaType.</returns>
+    /// <exception cref="ArgumentException"></exception>
+    /// <exception cref="NotSupportedException"></exception>
+    internal static (string, string) GenerateImageIndex(ImageInfo[] imageInfos)
```

Comment:
yes, because I'm checking for the `ManifestMediaType` of the first imageInfo to make a choice of creating docker manifest list, or oci image index. if there are no images then there is no point of creating image index